### PR TITLE
issue #79 aesthetics of example and __main__ cleanup + more maintanable

### PR DESCRIPTION
**Pull request by PyMarc2**
-The output are in HTML
-tried to be aesthetically pleasing codewise and examplewise
-the examples can be written -as such- in exampleX.py and it will generate it automatically.

### CODE REVIEW

**Comment by PyMarc2** [comment]
Verify conflicts with .gitignore before merging...

File: raytracing/__main__.py [hunk 1]
```diff
@@ -1,430 +1,123 @@
...
+        fig, axes = plt.subplots(figsize=(7, 5))
+        self.figureObject.createRayTracePlot(axes=axes)
+        with tempfile.TemporaryFile(suffix=".png") as tmpfile:
+            plt.savefig(tmpfile, format="png")
+            tmpfile.seek(0)
+            self.base64Figure = base64.b64encode(tmpfile.read()).decode()
+        self.HTMLFigure = ''''<div style="display:inline-block; vertical-align:top"><img class="icon" src="data:image/png;base64,{} "></div>'''.format(
+            self.base64Figure)
+
+    def generateHTMLHighlightedCode(self, code=""):
```
Comment (JLBegin) [inline raytracing/__main__.py:81]: Do you plan on using this `code` argument ?

Comment (PyMarc2) [reply raytracing/__main__.py:81]: It was for future proofing. Daniel talked about inline string code input. Even though i'm not sure about that, I kept it in mind that the argument of -s might be a string to highlight

File: raytracing/__main__.py [hunk 2]
```diff
@@ -1,430 +1,123 @@
...
+        code = self.getExampleCode()
+        docString = self.getDocstring(code)
+        self.HTMLDescription = '''<div class="alert alert-success" role="alert"><h4 class="alert-heading">Example Description</h4><div style="word-wrap: break-word;
+   width: 600px;">{}</div></div>'''.format(docString)
+
+    def runExampleCode(self):
+        with patch('matplotlib.pyplot.show') as p:
+            global path
+            exec(open(self.selectedFile).read(), globals())
+            self.figureObject = path
```
Comment (JLBegin) [inline raytracing/__main__.py:99]: I don't think we should call this ImagingPath a figureObject since it's not.

Comment (PyMarc2) [reply raytracing/__main__.py:99]: very true.

File: raytracing/__main__.py (hunk 1 above)
Comment (JLBegin) [reply raytracing/__main__.py:81]: Ok. I guess this could enable a user to create it's own html examples...  in the future

**Comment by PyMarc2** [comment]
Discovered error. in __main__ we arg is 'all', must make the index = i +1

File: raytracing/__main__.py [hunk 3]
```diff
@@ -1,430 +1,123 @@
-from .imagingpath import *
-from .laserpath import *
+import argparse
+from pygments import highlight
+import pygments.lexers
+import pygments.formatters
+import webbrowser
+from unittest.mock import patch
+from matplotlib import pyplot as plt
+from pylatexenc import latexencode
```
Comment (JLBegin) [inline raytracing/__main__.py:8]: requires package `pylatexenc`

Comment (PyMarc2) [reply raytracing/__main__.py:8]: it does not, my bad

File: raytracing/__main__.py [hunk 4]
```diff
@@ -1,430 +1,123 @@
...
-from .axicon import *
-import raytracing.thorlabs as thorlabs
-import raytracing.eo as eo
-import raytracing.olympus as olympus
 
-import argparse
 ap = argparse.ArgumentParser(prog='python -m raytracing')
-ap.add_argument("-e", "--examples", required=False, default='all', help="Specific example numbers, separated by a comma")
-
+ap.add_argument("-e", "--examples", required=False, default='', help="Specific example numbers, separated by a comma")
```
Comment (JLBegin) [inline raytracing/__main__.py:15]: We could specify the allowed range, and catch a valueError if out of range

**Comment by PyMarc2** [comment]
- The only non native new package is pygments, which is necessary anyway if the code has to be highlighted.
- It should now run because the pylatex module wasn't supposed to be there
- you believe it is less readable? sad, I think it is much better than 30 example in one document!
- yes the OS dependency is a problem to verify.
- yes I thought you would want the output to be in native matplotlib display
- I thought the examples gave exactly the same result as before, not sure I understand ?

**Comment by dccote** [comment]
> * you believe it is less readable? sad, I think it is much better than 30 example in one document!

The `__main__` is less readable. Clean it up big time to quickly get the code from all example files in a simple loop on example dir.  We will also call the files with useful names.  It could be example-kohlerIllumination.py for instance.
We may want to move the folder inside raytracing, and provide the link to the user so they can open the directory.

I think it's important that the example file be standalone.  `__main__` will simply loop through them.

**Comment by dccote** [comment]
We'll revisit this later, for now, let's close the Pull Request and keep the branch.

**Pull request closed without merging**